### DISCUSSION TITLE
fail to build from source

### DISCUSSION THREAD

**Discussion by WT215** [jax-ml/jax#11122] [Q&A]
Following the instruction from https://jax.readthedocs.io/en/latest/developer.html:
Not sure if it is because CPU is not suitable?
```
$ python build/build.py

     _   _  __  __
    | | / \ \ \/ /
 _  | |/ _ \ \  /
| |_| / ___ \/  \
 \___/_/   \/_/\_\


Bazel binary path: ./bazel-5.1.1-linux-x86_64
Bazel version: 5.1.1
Python binary path: /home/ma/w/sadfwewfasd/anaconda3/envs/cell2loc/bin/python
Python version: 3.9
NumPy version: 1.22.4
MKL-DNN enabled: yes
Target CPU: x86_64
Target CPU features: release
CUDA enabled: no
TPU enabled: no
ROCm enabled: no

Building XLA and installing it in the jaxlib source tree...
./bazel-5.1.1-linux-x86_64 run --verbose_failures=true --config=avx_posix --config=mkl_open_source_only :build_wheel -- --output_path=/home/clustor2/ma/w/wt215/jax/dist --cpu=x86_64
FATAL: unexpected result from F_OFD_SETLK: (error: 5): Input/output error
b''
Traceback (most recent call last):
  File "/home/clustor2/ma/w/wt215/jax/build/build.py", line 528, in <module>
    main()
  File "/home/clustor2/ma/w/wt215/jax/build/build.py", line 523, in main
    shell(command)
  File "/home/clustor2/ma/w/wt215/jax/build/build.py", line 53, in shell
    output = subprocess.check_output(cmd)
  File "/home/ma/w/wt215/anaconda3/envs/cell2loc/lib/python3.9/subprocess.py", line 424, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/home/ma/w/wt215/anaconda3/envs/cell2loc/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['./bazel-5.1.1-linux-x86_64', 'run', '--verbose_failures=true', '--config=avx_posix', '--config=mkl_open_source_only', ':build_wheel', '--', '--output_path=/home/clustor2/ma/w/wt215/jax/dist', '--cpu=x86_64']' returned non-zero exit status 36.

```

**Comment by DerToast3r** [comment]
Hi, I had trouble compiling it for an Xeon x5670 with cuda and cudnn support, but the compiled code did fail at some unknown variable..
The flag " --target_cpu_features='native' " did solve some issues for me.

**Reply by WT215** [reply]
I ended up at another HPC platform (try to install [cell2loc](https://github.com/BayraktarLab/cell2location), but failed to import dependent `jax` when import that package). Thank you anyway!

**Reply by hawkinsp** [reply]
@DerToast3r if this is something you would like us to look into, please open an issue with instructions to reproduce the problem. (Odds  are it means you have an old/incompatible compiler, but we can try to double check that.)

**Comment by hawkinsp** [comment]
For the original question about `F_OFD_SETLK`, I strongly suspect that means that you are building from (or some of Bazel's temporary outputs are being placed on) a network filesystem of some type. That apparently does not work. All I can suggest "build from a POSIX-compliant filesystem, e.g., a local disk or tmpfs".
